src/KOKKOS/bond_hybrid_kokkos.cpp

```diff
@@ -0,0 +1,214 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   LAMMPS development team: developers@lammps.org
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "bond_hybrid_kokkos.h"
+
+#include "atom_kokkos.h"
+#include "atom_masks.h"
+#include "comm.h"
+#include "error.h"
+#include "force.h"
+#include "kokkos.h"
+#include "memory_kokkos.h"
+#include "neighbor_kokkos.h"
+
+#include <cstring>
+
+using namespace LAMMPS_NS;
+
+#define EXTRA 1000
+
+/* ---------------------------------------------------------------------- */
+
+BondHybridKokkos::BondHybridKokkos(LAMMPS *lmp) : BondHybrid(lmp)
+{
+  kokkosable = 1;
+
+  atomKK = (AtomKokkos *) atom;
+  neighborKK = (NeighborKokkos *) neighbor;
+
+  execution_space = Device;
+
+  datamask_read = EMPTY_MASK;
+  datamask_modify = EMPTY_MASK;
+}
+
+/* ---------------------------------------------------------------------- */
+
+BondHybridKokkos::~BondHybridKokkos()
+{
+  deallocate();
+}
+
+/* ---------------------------------------------------------------------- */
+
+void BondHybridKokkos::compute(int eflag, int vflag)
+{
+  // save ptrs to original bondlist
+
+  int nbondlist_orig = neighbor->nbondlist;
+  neighborKK->k_bondlist.sync_device();
+  auto k_bondlist_orig = neighborKK->k_bondlist;
+  auto d_bondlist_orig = k_bondlist_orig.d_view;
+  auto d_nbondlist = k_nbondlist.d_view;
+  auto h_nbondlist = k_nbondlist.h_view;
+
+  // if this is re-neighbor step, create sub-style bondlists
+  // nbondlist[] = length of each sub-style list
+  // realloc sub-style bondlist if necessary
+  // load sub-style bondlist with 3 values from original bondlist
+
+  if (neighbor->ago == 0) {
+    Kokkos::deep_copy(d_nbondlist,0);
+
+    k_map.sync_device();
+    auto d_map = k_map.d_view;
+
+    Kokkos::parallel_for(nbondlist_orig,LAMMPS_LAMBDA(int i) {
+      const int m = d_map[d_bondlist_orig(i,2)];
+      if (m >= 0) Kokkos::atomic_increment(&d_nbondlist[m]);
+    });
+
+    k_nbondlist.modify_device();
+    k_nbondlist.sync_host();
+
+    for (int m = 0; m < nstyles; m++)
+      if (h_nbondlist[m] > maxbond_all)
```

Comment:
Is `maxbond_all` initialized anywhere?

Reply:
@jtclemm good catch, not sure how I missed this.